packages/cli/docs-resolver/src/DocsDefinitionResolver.ts

```diff
@@ -624,11 +624,168 @@ export class DocsDefinitionResolver {
+                        ].filter(isNonNullish)
+                    }),
+                    ...(snippets.node && {
+                        node: [...(existing.node ?? []), ...(snippets.node ?? [])].filter(isNonNullish)
+                    }),
+                    ...(snippets.ruby && {
+                        ruby: [...(existing.ruby ?? []), ...(snippets.ruby ?? [])].filter(isNonNullish)
+                    }),
+                    ...(snippets.typescript && {
+                        typescript: [...(existing.typescript ?? []), ...(snippets.typescript ?? [])].filter(
+                            isNonNullish
+                        )
+                    })
+                });
+
+                const createInitialSnippets = (
+                    snippets: Record<FdrAPI.api.latest.Language, FdrAPI.api.latest.CodeSnippet[] | undefined>
+                ): Record<FdrAPI.api.latest.Language, FdrAPI.api.latest.CodeSnippet[]> => ({
+                    ...(snippets.go && { go: snippets.go }),
+                    ...((snippets.pythonAsync || snippets.pythonSync) && {
+                        python: [...(snippets.pythonAsync ?? []), ...(snippets.pythonSync ?? [])].filter(isNonNullish)
+                    }),
+                    ...(snippets.node && { node: snippets.node }),
+                    ...(snippets.ruby && { ruby: snippets.ruby }),
+                    ...(snippets.typescript && { typescript: snippets.typescript })
+                });
+
+                Object.values({ _: fernApi.rootPackage, ...fernApi.subpackages }).forEach((package_) =>
+                    package_.endpoints.forEach((fernEndpoint) => {
+                        const matchingOpenApiEndpoint = Object.values(api.endpoints).find(
+                            (openApiEndpoint) =>
+                                openApiEndpoint.method === fernEndpoint.method &&
+                                openApiEndpoint.path.join("/") === fernEndpoint.path.parts.join("/")
+                        );
+
+                        if (matchingOpenApiEndpoint) {
+                            fernEndpoint.examples.forEach((fernExample) => {
+                                const matchingOpenApiExample = matchingOpenApiEndpoint.examples?.find(
+                                    (e) => e.name === fernExample.name
+                                );
+
+                                if (matchingOpenApiExample) {
+                                    const snippets = createSnippetsForExample(fernExample);
+                                    matchingOpenApiExample.snippets = matchingOpenApiExample.snippets
+                                        ? mergeSnippets(matchingOpenApiExample.snippets, snippets)
+                                        : createInitialSnippets(snippets);
+                                }
+                            });
+                        }
+                    })
+                );
+            };
+
+            mergeApiExamples(fernApi, api);
```

Comment:
what do you think about moving all of this into a separate file called `mergeIRExamplesWithFdrLatestExamples` ?